// Copyright (c) Authors of Clover
//
// All rights reserved. This program and the accompanying materials
// are made available under the terms of the Apache License, Version 2.0
// which accompanies this distribution, and is available at
// http://www.apache.org/licenses/LICENSE-2.0


syntax = "proto3";
package snort;

service Controller {
  
  rpc AddRules (AddRule) returns (SnortReply) {}
  rpc StartSnort (ControlSnort) returns (SnortReply) {}
  rpc StopSnort (ControlSnort) returns (SnortReply) {}
}

message ControlSnort {
  string pid = 1;
}

message AddRule {
  string protocol = 1;
  string dest_port = 2;
  string dest_ip = 3;
  string src_port = 4;
  string src_ip = 5;
  string msg = 6;
  string sid = 7;
  string rev = 8;
}

message SnortReply {
  string message = 1;
}
